### PR TITLE
Add arm32 Dockerfile

### LINKED ISSUE #4931: docker image for ARM
Since we do have ARM build, why not an ARM docker image?

The official images [arm32v7/golang](https://hub.docker.com/r/arm32v7/golang/) and [arm32v7/busybox](https://hub.docker.com/r/arm32v7/busybox/) can be used as the base image.

Moreover, `COPY --from=0 /lib/x86_64-linux-gnu/libdl-2.24.so /lib/libdl.so.2` should be changed to `COPY --from=0 /lib/arm-linux-gnueabihf/libdl-2.24.so /lib/libdl.so.2` and `  && wget -q -O tini https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini \` should be changed to `  && wget -q -O tini https://github.com/krallin/tini/releases/download/$TINI_VERSION/tini-armhf \`

I've successfully built images [here](https://hub.docker.com/r/yrzr/go-ipfs-arm64v8).

@lgierth

### PR DESCRIPTION
**Pull request by johnelliott**
fixes #4931 

...or just related, looking for assistance form maintainers for what additional work to do here to maintain the image

### CODE REVIEW

**Comment by johnelliott** [comment]
According to @yrzr [here](https://github.com/ipfs/go-ipfs/issues/4931#issue-313207104) and my experience building, that was the only thing required. They [mentioned](https://github.com/ipfs/go-ipfs/issues/4931#issuecomment-410568420) that it is a convention, however given the small difference I was wondering the same thing.

**Comment by johnelliott** [comment]
I've looked at some ways to parameterize the Dockerfile but haven't begun work on it.

**Comment by johnelliott** [comment]
Side note here:

The GitCop 'See the guidelines about this' link to https://github.com/ipfs/community/blob/master/contribution-guidelines.md#a-license-and-a-signed-off-by-trailers-are-required is a 404.

**Comment by lgierth** [comment]
Nice, this looks good to me!

The next steps are documenting the build args in the Dockerfiles themselves and in the README.md docker section.

And maybe (later) we can have Makefile targets a la `make docker_amd64` and `make docker_armv7`.

**Comment by johnelliott** [comment]
OK, I will first see if a convention exists for such documentation and add some notes. I see the existing comment about maintaining the faster-building file. I'll update the readme also.

**Comment by johnelliott** [comment]
I added documentation of the argument and some notes to the README with example build commands

**Comment by johnelliott** [comment]
Looks like I'll need to give this some more attention until a maintainer can review and merge this.

**Comment by doodlemania2** [comment]
any update on this? looking for an ipfs docker container for armhf

**Comment by yrzr** [comment]
@doodlemania2 Seems they haven't added the docker image yet. You can use my build for [arm32v7](https://hub.docker.com/r/yrzr/go-ipfs-arm32v7/) and [arm64v8](https://hub.docker.com/r/yrzr/go-ipfs-arm64v8/) first. Or you can build your own image following #4931

**Comment by johnelliott** [comment]
@doodlemania2 I got started with @yrzr's work and this PR is based on his work. Your fastest path forward is using their builds. I used them successfully before starting this branch.

**Comment by doodlemania2** [comment]
Thanks all!

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,8 @@
-FROM golang:1.11-stretch
+# Go base image
+ARG base=golang:1.11-stretch
+# Busybox base image
+ARG busybox_base=busybox:1-glibc
```
Comment (Stebalien) [inline Dockerfile]: I'm not a docker expert, but shouldn't this just go down at line 46? Is there a reason to duplicate it?

File: README.md [hunk 2]
```diff
@@ -317,6 +317,48 @@ Stop the running container:
 
     docker stop ipfs_host
 
+### Building Docker images
```
Comment (Stebalien) [inline README.md]: Should be updated for go 1.11.

File: README.md [hunk 3]
```diff
@@ -317,6 +317,48 @@ Stop the running container:
...
+
+#### `base`
+Possible values:
+- `golang:1.10-stretch` (default)
+- `arm32v7/golang:1.10-stretch`
+
+#### `busybox_base`
+Possible values:
+- `busybox:1-glibc` (default)
+- `arm32v7/golang:1.10-stretch`
```
Comment (Stebalien) [inline README.md]: busybox.

File: README.md [hunk 4]
```diff
@@ -317,6 +317,48 @@ Stop the running container:
...
 
+### Building Docker images
+
+For other platforms such as ARM, you may need to build a docker image to run.
+
+The main Dockerfile takes the following args:
+- `base`: The base go image for the target platform
+- `busybox_base`: The busybox base image for the target platform
+- `tini_executable`: The tini executable path for the target platform
+- `glibc_shared_lib_arch`: The target platform shared lib for glibc supplementing busybox
```
Comment (Stebalien) [inline README.md]: Is there really no better way to do this (i.e., with one variable). From what I can tell, there simply isn't but I'm not a Docker expert so I'd like to double check.

Comment (johnelliott) [reply README.md]: Same here. This was seemingly the only way.

**Comment by johnelliott** [comment]
I may get to this this week :)

**Comment by johnelliott** [comment]
There doesn't seem to be sufficient desire for this change yet so I'm closing this PR. Anyone else can feel free to take it up again :)

**Comment by Stebalien** [comment]
Yeah, sorry about that. This looks useful but nobody (that I know of) on the current team has extensive knowledge of docker or the need to run go-ipfs on arm. That makes this a bit difficult to validate, test, and maintain.

However, if someone has a real need for this, please comment and we can reopen (or continue in a new PR).

**Comment by doodlemania2** [comment]
Thanks all - I ended up rolling my own: https://github.com/doodlemania2/ethereum-armhf based on other's work.

**Comment by master-hax** [comment]
btw the go-ipfs docker images provided by linuxserver.io work on armhf: https://hub.docker.com/r/linuxserver/ipfs

**Pull request closed without merging**